xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/layouts/LayoutLoader.vue

```diff
@@ -0,0 +1,129 @@
+<!--
+  * See the NOTICE file distributed with this work for additional
+  * information regarding copyright ownership.
+  *
+  * This is free software; you can redistribute it and/or modify it
+  * under the terms of the GNU Lesser General Public License as
+  * published by the Free Software Foundation; either version 2.1 of
+  * the License, or (at your option) any later version.
+  *
+  * This software is distributed in the hope that it will be useful,
+  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  * Lesser General Public License for more details.
+  *
+  * You should have received a copy of the GNU Lesser General Public
+  * License along with this software; if not, write to the Free
+  * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+  * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ -->
+
+
+<template>
+  <div
+    :class="['layout-loader', { waiting: loading }]"
+  >
+    <div class="loader-fill"></div>
+  </div>
+</template>
+
+
+<script>
+export default {
+
+  name: "layout-loader",
+
+  inject: ["logic"],
+
+  data () {
+    return {
+      loadingData: false,
+      loadingLayout: false,
+      delay: 200,
+    };
+  },
+
+  computed: {
+    loading () {
+      return this.loadingData || this.loadingLayout;
+    }
+  },
+
+  methods: {
+    watchLoading (propertyName, eventBefore, eventAfter) {
+      let timeoutId;
+
+      this.logic.onEvent(eventBefore, () => {
+        console.log("BEFORE", propertyName, eventBefore, eventAfter);
+        clearTimeout(timeoutId);
+        timeoutId = setTimeout(() => {
+          this[propertyName] = true;
+        }, this.delay);
```

Comment:
Why do you wait 200ms to mark the loading state? This needs an explanation. Note that this approach makes writing functional tests a pain. It's fine for changing the layout because there are some CSS classes to wait for (``layout-cards``) but for data it's different. If you write a test that checks the pagination you have this situation:

* the test clicks on the next page link and needs to wait for something that says the live data is ready
* if you wait for ``waiting`` class to disappear then it may return before the loading starts (because the live data is not waiting right after clicking on the next page link)
* if you wait for ``waiting`` class to appear and then disappear then this may happen too fast and the test will just see no waiting class and thus fail

If you don't want to show the loading animation right away, I'm fine with that, but the live data should be marked as loading right away (in the HTML). So you may need 2 classes for this (one to mark the loading state and one to show it to the user).